applications/asset_tracker_v2/src/modules/cloud_module.c

```diff
@@ -324,6 +324,38 @@ static void pgps_data_handle(const uint8_t *buf, const size_t len)
 #endif /* CONFIG_NRF_CLOUD_PGPS */
 }
 
+static void cloud_location_data_handle(uint8_t *buf, const size_t len)
+{
+	int err;
+	struct location_data location;
+	struct cloud_module_event *cloud_module_event;
+
+	err = cloud_codec_decode_cloud_location(buf, len, &location);
+	if (err == 0) {
+		cloud_module_event = new_cloud_module_event();
```

Comment:
Some reaping code, perhaps something like this would do good:

```
void submit_cloud_module_event(enum cloud_module_event_type type, int error) {
    struct cloud_module_event *cloud_module_event = new_cloud_module_event();
    __ASSERT(cloud_module_event, "Not enough heap left to allocate event");
    
    cloud_module_event->type = type;
    if (type == CLOUD_EVT_CLOUD_LOCATION_RECEIVED) {
        cloud_module_event->data.cloud_location = location;
    } else if (type == CLOUD_EVT_ERROR) {
        cloud_module_event->data.error = error;
    }
    
    APP_EVENT_SUBMIT(cloud_module_event);
}

err = cloud_codec_decode_cloud_location(buf, len, &location);
if (err == 0) {
    submit_cloud_module_event(CLOUD_EVT_CLOUD_LOCATION_RECEIVED, 0);
} else if (err == -EFAULT) {
    submit_cloud_module_event(CLOUD_EVT_CLOUD_LOCATION_ERROR, err);
} else if (err == -ENOTSUP) {
    submit_cloud_module_event(CLOUD_EVT_CLOUD_LOCATION_UNKNOWN, err);
} else {
    LOG_ERR("Decoding of cloud location response, error: %d", err);
    SEND_ERROR(cloud, CLOUD_EVT_ERROR, err);
}
```